lib/services/android-project-service.ts

```diff
@@ -118,19 +119,135 @@ class AndroidProjectService implements IPlatformProjectService {
 			return assetsDirectory;
 
 		}).future<string>()();
-	}
+    }
+
+    private updateMetadata(projectRoot: string): void {
+        var projMetadataDir = path.join(projectRoot, "assets", "metadata");
+        var libsmetadataDir = path.join(projectRoot, "../../lib", this.platformData.normalizedPlatformName, "__metadata");
+        shell.cp("-f", path.join(libsmetadataDir, "*.dat"), projMetadataDir);
+    }
+
+    private generateMetadata(projectRoot: string): void {
+        var metadataGeneratorPath = path.join(__dirname, "../../resources/tools/metadata-generator.jar");
+        var libsFolder = path.join(projectRoot, "../../lib", this.platformData.normalizedPlatformName);
+        var metadataDirName = "__metadata";
+        var inputDir = libsFolder;
+        var outDir = path.join(libsFolder, metadataDirName);
+        this.$fs.ensureDirectoryExists(outDir).wait();
+
+        shell.cp("-f", path.join(__dirname, "../../resources/tools/android.jar"), inputDir);
+        shell.cp("-f", path.join(__dirname, "../../resources/tools/android-support-v4.jar"), inputDir);
+        shell.cp("-f", path.join(projectRoot, "libs/*.jar"), inputDir);
+
+        this.spawn('java', ['-jar', metadataGeneratorPath, inputDir, outDir]).wait();
+    }
 
 	public buildProject(projectRoot: string): IFuture<void> {
-		return (() => {
+        return (() => {
 			var buildConfiguration = options.release ? "release" : "debug";
-			var args = this.getAntArgs(buildConfiguration, projectRoot);
-			this.spawn('ant', args).wait();
+            var args = this.getAntArgs(buildConfiguration, projectRoot);
+            var args2 = this.getAntArgs(buildConfiguration, projectRoot);
```

Comment:
rename variable `args2` to `argsSaved` as suggested https://github.com/fealebenpae/nativescript-cli/commit/ff1273404ad1225163fde0d4982f075b363cfe0d